### PR TITLE
fix(clippy): resolve various clippy warnings

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

Clippy has some new warnings on nightly, and we also have some warnings in CI on non-linux platforms.

Depends-On: #4483

## Solution

Potential performance/locking changes:
- move cached root dereference earlier, so the lock isn't held for the entire match statement

Trivial changes:
- fix unused import warning on non-linux platforms
- derive `Eq` where we already derive `PartialEq`

We should run all the tests to check for any breaking changes.

## Review

This PR is a low priority.

### Reviewer Checklist

  - [ ] Tests for Expected Behaviour
  - [ ] Tests for Errors

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -47,6 +47,13 @@ rustflags = [
     # Code styles we want to accept
     "-Aclippy::try_err",
 
+    # Lints we need to ignore
+    #
+    # TODO: re-enable when:
+    # - false positives are fixed, and
+    # - prost fixes these lints in generated code
+    "-Aclippy::derive_partial_eq_without_eq",
```
Comment (teor2345) [inline .cargo/config.toml]:
```suggestion
    "-Aclippy::derive_partial_eq_without_eq",
    # Rust stable doesn't know about the above lint name yet
    "-Aunknown_lints",
```

**Comment by conradoplg** [comment]
`time="2022-05-24T11:42:06Z" level=error msg="error waiting for container: unexpected EOF"` errors, retrying...

https://github.com/ZcashFoundation/zebra/runs/6570363128?check_suite_focus=true
https://github.com/ZcashFoundation/zebra/runs/6570363511?check_suite_focus=true

**Review by oxarbitrage — APPROVED** [review]
Looks good. This will also get rid of some annoying `color_eyre` warning that we have in every PR since a while.

**Review by dconnolly — APPROVED** [review]

**Comment by gustavovalverde** [comment]
@Mergifyio update

**Comment by teor2345** [comment]
> `time="2022-05-24T11:42:06Z" level=error msg="error waiting for container: unexpected EOF"` errors, retrying...
> 
> https://github.com/ZcashFoundation/zebra/runs/6570363128?check_suite_focus=true https://github.com/ZcashFoundation/zebra/runs/6570363511?check_suite_focus=true

@gustavovalverde are we accidentally deleting the wrong machines again?
This looks similar to #3991.

**Comment by gustavovalverde** [comment]
@teor2345 that was a manual deletion from GCP, not related to the pipeline. This PR had over 8 VMS running at the same time.

This issue was fixed with https://github.com/ZcashFoundation/zebra/commit/9335f68409f5616606e0dde7dc05205999cd7d0b

**Comment by teor2345** [comment]
@mergifyio update

**Review by upbqdn — APPROVED** [review]

**Comment by gustavovalverde** [comment]
@Mergifyio update

**Pull request merged**